cmd/kind/kind.go

```diff
@@ -20,40 +20,76 @@ package kind
 import (
 	"os"
 
+	"github.com/sirupsen/logrus"
+
 	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 
 	"sigs.k8s.io/kind/cmd/kind/build"
 	"sigs.k8s.io/kind/cmd/kind/create"
 	"sigs.k8s.io/kind/cmd/kind/delete"
+	logutil "sigs.k8s.io/kind/pkg/log"
 )
 
+const defaultLevel = logrus.WarnLevel
+
+// Flags for the kind command
+type Flags struct {
+	LogLevel string
+}
+
 // NewCommand returns a new cobra.Command implementing the root command for kind
 func NewCommand() *cobra.Command {
+	flags := &Flags{}
 	cmd := &cobra.Command{
 		Use:   "kind",
 		Short: "kind is a tool for managing local Kubernetes clusters",
 		Long:  "kind creates and manages local Kubernetes clusters using Docker container 'nodes'",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			return runE(flags, cmd, args)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return cmd.Help()
+		},
 	}
+	cmd.PersistentFlags().StringVar(&flags.LogLevel, "loglevel", defaultLevel.String(), "logrus log level")
 	// add all top level subcommands
 	cmd.AddCommand(build.NewCommand())
 	cmd.AddCommand(create.NewCommand())
 	cmd.AddCommand(delete.NewCommand())
 	return cmd
 }
 
+func runE(flags *Flags, cmd *cobra.Command, args []string) error {
+	level := defaultLevel
+	parsed, err := log.ParseLevel(flags.LogLevel)
+	if err != nil {
+		log.Warnf("Invalid log level '%s', defaulting to '%s'", flags.LogLevel, level)
```

Comment:
nit: May be worth doing {something} to ensure this message is always visible? `defaultLevel` is currently `warn`, but I'm slightly concerned if this gets changed in future 🙄

Reply:
Hmm, we could just make this yet another thing that does not actually use logging, I almost wound up writing our own small logging package :-)

Reply:
we're discussing broader logging options out of band, but this seems to be fine for now.